src/Optimization/hiopKKTLinSysSparse.cpp

```diff
@@ -580,59 +578,58 @@ namespace hiop
           linSys_ = p;        
 #endif  // HIOP_USE_STRUMPACK        
         }
-      }else{
-
-        //
-        // on device: compute_mode is hybrid, auto, or gpu
-        //
-        assert(nullptr==linSys_);
-
-        //At this point the only supported GPU sparse solver is STRUMPACK. 
-        //We expect new GPU sparse solvers to be added. These should take precedence over STRUMPACK
-        //in the instantiation logic below whenever it is ambiguous which GPU solver to instantiate,
-        //for example when "linear_solver_sparse" option is auto.
+      } else {
+      //we are on the GPU. Our first choice is always cuSolver
+#if  defined(HIOP_USE_CUSOLVER)        
+        hiopLinSolverIndefSparseCUSOLVER *p = new hiopLinSolverIndefSparseCUSOLVER(n, nnz, nlp_);
+        auto verbosity = hovScalars;
+        nlp_->log->printf(verbosity,
+            "KKT_SPARSE_XDYcYd linsys: alloc CUSOLVER size %d (%d cons) (safe_mode=%d)\n",
+            n, neq+nineq, safe_mode_);
+        if(safe_mode_) verbosity  = hovWarning;
 
-#ifdef HIOP_USE_STRUMPACK   
+        p->setFakeInertia(neq + nineq);
+        linSys_ = p;
+#else
+#if defined(HIOP_USE_STRUMPACK)
         hiopLinSolverIndefSparseSTRUMPACK *p = new hiopLinSolverIndefSparseSTRUMPACK(n, nnz, nlp_);
-
-        //print it as a warning if safe mode is on
-        auto verbosity = safe_mode_ ? hovWarning : hovScalars;
-
+        auto verbosity = hovScalars;
```

Comment:
It's not obvious to me what type `auto` means here - I would rather this be an explicit type.